src/pool/srv_target.c

```diff
@@ -1344,3 +1344,53 @@ ds_pool_tgt_prop_update(struct ds_pool *pool, struct pool_iv_prop *iv_prop)
 	pool->sp_reclaim = iv_prop->pip_reclaim;
 	return 0;
 }
+
+/**
+ * Query the cached pool map. If the cached version is <= in->tmi_map_version,
+ * reply -DER_ALREADY.
+ */
+void
+ds_pool_tgt_query_map_handler(crt_rpc_t *rpc)
+{
+	struct pool_tgt_query_map_in   *in = crt_req_get(rpc);
+	struct pool_tgt_query_map_out  *out = crt_reply_get(rpc);
+	struct ds_pool_hdl	       *hdl;
+	struct ds_pool		       *pool;
+	struct pool_buf		       *buf;
+	unsigned int			version;
+	int				rc;
+
+	D_DEBUG(DB_TRACE, DF_UUID": handling rpc %p\n",
+		DP_UUID(in->tmi_op.pi_uuid), rpc);
+
+	hdl = ds_pool_hdl_lookup(in->tmi_op.pi_hdl);
+	if (hdl == NULL) {
+		rc = -DER_NO_HDL;
+		goto out;
+	}
+
+	/* Inefficient; better invent some zero-copy IV APIs. */
```

Comment:
probably check or assert to make sure sp_map is not NULL here.

Reply:
Good point. Not checking that is definitely a bug.